osu.Game/Overlays/BeatmapListing/BeatmapSearchFilter.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.BeatmapListing
+{
+    public class BeatmapSearchFilter<T> : TabControl<T>
+    {
+        public BeatmapSearchFilter()
+        {
+            AutoSizeAxes = Axes.Both;
+
+            if (typeof(T).IsEnum)
+            {
+                foreach (var val in (T[])Enum.GetValues(typeof(T)))
+                    AddItem(val);
+            }
+        }
+
+        protected override Dropdown<T> CreateDropdown() => null;
+
+        protected override TabItem<T> CreateTabItem(T value) => new FilterTabItem(value);
+
+        protected override TabFillFlowContainer CreateTabFlow() => new TabFillFlowContainer
+        {
+            AutoSizeAxes = Axes.Both,
+            Spacing = new Vector2(10),
+        };
+
+        protected class FilterTabItem : TabItem<T>
+        {
+            protected virtual float TextSize() => 13;
```

Comment:
A little un-C#-like to make this a method instead of a getter-only property...?

```suggestion
            protected virtual float TextSize => 13;
```

(requires fixing the override too)